xmppserver/src/main/java/org/jivesoftware/openfire/ldap/LdapGroupProvider.java

```diff
@@ -124,7 +133,67 @@ private Group getGroupByDN(LdapName groupDN, Set<String> membersToIgnore) throws
             ctx = manager.getContext(baseDN);
             Attributes attrs = ctx.getAttributes(relativeDN, standardAttributes);
 
-            return processGroup(ctx, attrs, membersToIgnore);
+            boolean paging = false;
+            int countPage=0;
+
+            NamingEnumeration<? extends Attribute> i = attrs.getAll();
+            String groupMemberField = manager.getGroupMemberField();
+            String searchString = groupMemberField.contains(";range=")?groupMemberField.substring(0,groupMemberField.indexOf(";range=")):groupMemberField;
+            while (i.hasMore())
+            {
+                Attribute attribute = i.next();
+                String key = attribute.getID().toLowerCase();
+                if (key.startsWith(searchString))
+                {
+                    paging=true;
```

Comment:
If I understand this correctly, then paging will only occur when an administrator explicitly sets the group member field to include `;range=`  I don't like that. Can't we make sure that paging happens, when needed, instead? That way, administrators need not be aware of a problem to be able to fix it.

Reply:
i will look into it tomorrow... but i think there was a problem that we have to use "member;range=0-*" as value for ldap.groupMemberFeld otherwise AD wont five any results for that big group or max 500...

Reply:
there is a 1500 limit in AD so read this:
https://docs.microsoft.com/de-de/previous-versions/windows/desktop/ldap/searching-using-range-retrieval?redirectedfrom=MSDN

because of this i created this PR...

Reply:
I'm not saying that we should not add the range attribute to the request from Openfire to AD. I do wonder if we can hide this complexity from the administrator. Instead of requiring the administrator to add this parameter manually, it would be nice if Openfire could do it automatically, if needed.

Reply:
then we should always use the member;range=0-* as default attribute for AD

Reply:
but i dont know if this limit is only a problem with AD or a general LDAP issue